Geometry/TrackerNumberingBuilder/src/GeometricDet.cc

```diff
@@ -124,7 +124,7 @@ GeometricDet::GeometricDet(const PGeometricDet::Item& onePGD, GeometricEnumType
            onePGD._a31,
            onePGD._a32,
            onePGD._a33),
-      _shape(static_cast<DDSolidShape>(onePGD._shape)),
+      _shape(static_cast<cms::DDSolidShape>(onePGD._shape)),
```

Comment:
I think this code is handling the persistent `PGeometricDet` from the DB. The object could be the old DD version with the old `DDSolidShape` value or the new version with `cms::DDSolidShape`, and I don't think there's any way to tell which one it is. The only safe approach is to make the integer values of the two `DDSolidShape`s identical.

Reply:
Yes, this was overlooked. I'll change the enum again to match the integer values between the two `DDSolidShape`s. However, that in itself is not enough because 7 and 8 (`ddpolycone_rrz` and `ddpolyhedra_rrz` don't exist for the DD4hep solidshape - and they should then map to `ddpolycone` and `ddpolyhedra` (ints 5 and 6). So I think on top of matching the enums, this line should always cast the `onePGD._shape` int  to a `LegacySolidShape` and then use the mapping functions to extract the corresponding `cms::DDSolidShape`. Am of course open to better suggestions.

Reply:
Yes, your approach is the safest way to do the conversion. In actuality, `ddpolycone_rrz` and `ddpolyhedra_rrz` are probably never used, but it is better to be safe.